controllers/vsl.go

```diff
@@ -109,53 +109,37 @@ func (r *DPAReconciler) LabelVSLSecrets(log logr.Logger) (bool, error) {
 
 func (r *DPAReconciler) ValidateVolumeSnapshotLocations(dpa oadpv1alpha1.DataProtectionApplication) (bool, error) {
 	for i, vslSpec := range dpa.Spec.SnapshotLocations {
+		vslYAMLPath := fmt.Sprintf("spec.snapshotLocations[%v]", i)
+		veleroVSLYAMLPath := vslYAMLPath + ".velero"
+		veleroConfigYAMLPath := "spec.configuration.velero"
+
 		if vslSpec.Velero == nil {
 			return false, errors.New("snapshotLocation velero configuration cannot be nil")
 		}
-		vsl := velerov1.VolumeSnapshotLocation{
-			ObjectMeta: metav1.ObjectMeta{
-				// TODO: Use a hash instead of i
-				Name:      fmt.Sprintf("%s-%d", r.NamespacedName.Name, i+1),
-				Namespace: r.NamespacedName.Namespace,
-			},
-			Spec: *vslSpec.Velero,
-		}
 
 		// check for valid provider
 		if vslSpec.Velero.Provider != AWSProvider && vslSpec.Velero.Provider != GCPProvider &&
-			vslSpec.Velero.Provider != Azure {
-			r.Log.Info("Non-supported provider specified, might be a misconfiguration")
-
-			r.EventRecorder.Event(&vsl,
-				corev1.EventTypeWarning,
-				"VSL provider is invalid",
-				fmt.Sprintf("VSL provider %s is invalid, might be a misconfiguration", vslSpec.Velero.Provider),
-			)
+			vslSpec.Velero.Provider != AzureProvider {
+			return false, fmt.Errorf("DPA %s.provider %s is invalid: only %s, %s and %s are supported", veleroVSLYAMLPath, vslSpec.Velero.Provider, AWSProvider, GCPProvider, AzureProvider)
```

Comment:
@sseago we talked about this, I think team preferred also allowing "namespace" plugins. With this change, `velero.io/aws` would be not allowed, only `aws`

This is already merged on master, should not block merge, but a follow up PR prior to 1.4.2 release may be required

Reply:
@mateusoliveira43 Yes, we should probably follow on with that change on master and 1.4 (keeping this commit the same as the merged master one).